File: content/examples/other/leetcode/problems/rotate-image/example-1/Solution.java
import java.util.Arrays;

public class Solution {
  static int[][] rotateImageExtraMatrix(int[][] matrix) {
    int n = matrix.length;
    int[][] out = new int[n][n];
    for (int r = 0; r < n; r += 1) {
      for (int c = 0; c < n; c += 1) out[c][n - 1 - r] = matrix[r][c];
    }
    for (int r = 0; r < n; r += 1) {
      for (int c = 0; c < n; c += 1) matrix[r][c] = out[r][c];
    }
    return matrix;
  }

  public static void main(String[] args) {
    int[][] matrix = new int[][] {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
    System.out.println(Arrays.deepToString(rotateImageExtraMatrix(matrix)));
  }
}

File: content/examples/other/leetcode/problems/rotate-image/example-1/solution.py
def rotate_image_extra_matrix(matrix):
    n = len(matrix)
    out = [[0] * n for _ in range(n)]
    for r in range(n):
        for c in range(n):
            out[c][n - 1 - r] = matrix[r][c]
    for r in range(n):
        for c in range(n):
            matrix[r][c] = out[r][c]
    return matrix


if __name__ == "__main__":
    print(rotate_image_extra_matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]]))
